### DISCUSSION TITLE
Using rye to create Py03 extension?

### DISCUSSION THREAD

**Discussion by BruceEckel** [astral-sh/rye#389] [General]
Py03 uses `maturin` which uses `pip`, but `rye` doesn't support `pip`.

**Comment by ischaojie** [comment]
> rye doesn't support pip

rye intentionally hides`pip`, which it uses internally.

**Reply by BruceEckel** [reply]
So can pip be exposed in some way so that `maturin` can use it for Py03?

**Reply by TillerBurr** [reply]
Can't you just add `pip` to the project dependencies?

**Comment by mitsuhiko** [comment]
I wrote docs for this now and improved support: https://rye-up.com/guide/rust/

**Comment by BruceEckel** [comment]
Thank you! That's excellent!

Here: "Likewise rye build will us maturin" I think you mean "use" rather than "us".

**Reply by mitsuhiko** [reply]
Thank you. Fixed!
